feat: add the git-hook:install command

AdeAttwood opened the pull request:
feat: add the git-hook:install command

Summary:

Create the `git-hook:install` command to install git hooks. This is the same as
the last one. We have added some extra environment variables to the hooks to
allow for future expansion. This will mainly give us the ability to prompt the
user to update the hooks if they are out of date.

Test Plan:

Tests have been written, I have also tested this manually by running the
command in a sapling and git repo.

Ref: #76

---
[//]: # (BEGIN SAPLING FOOTER)
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/Practically/Conventional-Tools/pull/85).
* #91
* #94
* #93
* #92
* __->__ #85

GeorgeCadwallader commented on tests/commands/git-hook:install.spec.ts:
```diff
@@ -0,0 +1,66 @@
+      }
+    });
+
+    it('dose not set the exit code because an error was thrown', ctx => {
```
**typo:** "dose" > "does"

```suggestion
    it('does not set the exit code because an error was thrown', ctx => {
```

AdeAttwood committed:
feat: add the git-hook:install command

Summary:

Create the `git-hook:install` command to install git hooks. This is the same as
the last one. We have added some extra environment variables to the hooks to
allow for future expansion. This will mainly give us the ability to prompt the
user to update the hooks if they are out of date.

Test Plan:

Tests have been written, I have also tested this manually by running the
command in a sapling and git repo.

Ref: #76

GeorgeCadwallader commented:
All looks good to me, just got some tests failing that need sorting :+1:
